src/npmUtils.ts

```diff
@@ -0,0 +1,50 @@
+import fs from "fs";
+import * as ini from "ini";
+
+export const prepareNpmConfig = (
+  // Allow to inject a custom object, useful in tests.
+  processEnv = process.env
+) => {
+  const userNpmrcPath = `${processEnv.HOME}/.npmrc`;
+
+  if (fs.existsSync(userNpmrcPath)) {
+    console.log(`Found existing user .npmrc file at ${userNpmrcPath}`);
+
+    // Parse the `.npmrc` content using the `npm/ini` package.
+    const npmConfig = ini.parse(fs.readFileSync(userNpmrcPath, "utf-8"));
```

Comment:
We use `ini.parse` to take care of parsing the config (all whitespaces are stripped out)